config/config.go

```diff
@@ -0,0 +1,163 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/googleinterns/aso_sxs_viewer/proto"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+type ViewerConfig struct {
+	proto.ViewerConfig
+}
+
+type BrowserConfig struct {
+	Selector string
+	Position int
+	URL      string
+}
+
+// GetConfig generates .aso_sxs_viewer directory and config file, if it doesn't already exists.
+// It validates the pre-existing config if found.
+// You should avoid repeated calls to avoid validation overhead.
+func GetConfig() (*ViewerConfig, error) {
+	if err := createDir(AsoSxSViewerDir); err != nil {
+		return nil, fmt.Errorf("Error %s encountered while creating aso_sxs_viewer directory", err.Error())
+	}
+
+	return createOrValidateConfig(AsoSxSViewerConfig)
+}
+
+func createDir(dirPath string) error {
+	// check if directory exists
+	_, err := os.Stat(dirPath)
+
+	// create directory if does not exist
+	if os.IsNotExist(err) {
+		return os.MkdirAll(dirPath, 0755)
+	}
+	return err
+}
+
+func createOrValidateConfig(filePath string) (*ViewerConfig, error) {
+	// check if file exists
+	_, err := os.Stat(filePath)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
+		file, err := os.Create(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
```

Comment:
This feels weird. You are calling close, but the file is not even open yet. The file is opened within generate config.
I think you should make generateconfig return the config, and move the file writing part to here.

Remember, returning the config rather than writing the config inside the function would also make it easier to test.